pkg/apis/batch/validation/validation.go

```diff
@@ -400,6 +405,27 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 	if status.Terminating != nil {
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*status.Terminating), fldPath.Child("terminating"))...)
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.JobManagedByLabel) {
```

Comment:
I don't really understand how these validations are related to this feature.

Maybe you can expand on that and add some comments?

Error messages should probably reflect that this is somehow related to this feature.

Reply:
> I don't really understand how these validations are related to this feature.

This is discussed in the KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-label-for-batch-jobs#job-status-validation. Basically, we want to mitigate the risk of external controllers doing "weird" things, that could break consumers of the Job API. There are some things which the built-in controller would not do (such as transitioning from Complete -> Failed), but other faulty controllers could.  In particular, we do that to ensure the assumptions inside the CronJob are satisfied.

> Maybe you can expand on that and add some comments?

Sure, let me know, I will go through the PR and add some, we can then discuss them individually in different threads.

> Error messages should probably reflect that this is somehow related to this feature.

I'm not sure that, I think the validation is required for the feature to constrain custom controllers, but it is valid in its own right. Maybe we can have it in the comments to say "// validate status to ensure custom controllers don't do unexpected things", but I'm not sure about error messages.

Reply:
Comments added, PTAL.